### ISSUE TITLE
[2.0] lost connexion with cloud

### ISSUE THREAD

**Issue by Raizo62** [GNS3/gns3-gui#1829]
Hi

On Linux

I connect a VMBox to ethernet switch. This ethernet switch is connected to Cloud device

The VMBox gets IP with DHCP of "Internet". The VMBox communicates with Internet.

I am waiting 10 minutes without using network in VMBox.

Then, VMBox can not communicate with Internet

Perhaps, the same bug as #1763

**Comment by noplay** [comment]
Hi,

Which kind of interface do you put in the cloud?
Do you see traffic on the link when you use wireshark?

**Comment by Raizo62** [comment]
euh....close....i can't reproduce....sorry

**Comment by Raizo62** [comment]
bug with cloud arrives often (almost all my student this morning), but i don't why.

I have this lab that works on a PC and bug on another. I don't know if this is the cause, but the PC with bug has 2 network cards.

vpcs can't obtain IP with DHCP : I see DHCPDISCOVER and DHCPOFFER on real card, but i see only DHCPDISCOVER on link with cloud.

Here is archive of all : [cloud_backup_170202_17_1352.tar.xz.txt](https://github.com/GNS3/gns3-gui/files/748380/cloud_backup_170202_17_1352.tar.xz.txt)
Here is debug export of GNS3 : [cloud.zip.txt](https://github.com/GNS3/gns3-gui/files/748382/cloud.zip.txt)

**Comment by Raizo62** [comment]
I have again the bug, but a simple lab on a PC with only one network card. 

When VM wants to ping, i see on the real card the broadcast request and reponse. But on the cloud, i see only the broadcast request.

I have no idea how works the cloud on linux then i don't know what i must verify or test. Can you explain me what i can do (or give me a link to the techno) ?

**Comment by Raizo62** [comment]
I have a little clue : 

When i lost the connection, the file 
projects/untitled/project-files/builtin/06ffc123-6523-4d24-a648-742137142b50/ubridge.log contains only :
`send: Bad address
`

**Comment by Raizo62** [comment]
> Which kind of interface do you put in the cloud?

Sorry, i forgot your question : i use "eth0"

**Comment by noplay** [comment]
We inject packet using the pcap library.

Do you have the issue if instead using DHCP you use a static IP? 

The eth0 is use only by GNS3 or do you have also real traffic on it?

**Comment by Raizo62** [comment]
> Do you have the issue if instead using DHCP you use a static IP?

Now, i don't know. I test static IP when the bug is already here

> The eth0 is use only by GNS3 or do you have also real traffic on it?

Often when bug, eth0 is used for access to vdi by nfs. In other case, i use browser web

**Comment by noplay** [comment]
When you say NFS it's the builtin share directory of Vbox?

**Comment by Raizo62** [comment]
>When you say NFS it's the builtin share directory of Vbox?

No. With Linux, i mount a remote directory with nfs. Then i must have traffic network
(But, Friday, i don't use it).

> When i lost the connection, the file
> untitled/project-files/builtin/06ffc123-6523-4d24-a648-742137142b50/ubridge.log contains only :
> send: Bad address

Can i have ubdrige more verbose, just for debug ? (compile option ?)

**Comment by noplay** [comment]
What is eth0 in Vbox it's a host only interface or a bridge?

And the IP of your VMbox connected to the cloud is not used somewhere else in the network?

**Comment by Raizo62** [comment]
> What is eth0 in Vbox it's a host only interface or a bridge?

VBox is the VMRaizo. 6 network interfaces, but only one is connected to the cloud (by Ethernet switch)
But i am not sure that i understand

> And the IP of your VMbox connected to the cloud is not used somewhere else in the network?

As the ethernet card is dhcp client, then i suppose that it is alone.
But perhaps that mac address (virtual) is not single (But I am not convinced)

**Comment by noplay** [comment]
I mean in Vbox the eth0 is attached to which type:
![image](https://cloud.githubusercontent.com/assets/345437/22684315/21b35f98-ed1c-11e6-9392-13c917b6d296.png)

**Comment by Raizo62** [comment]
I don't use config interface of VirtualBox. I configure only with GNS3. 
As the config IP with DHCP is in good network, i suppose it is correct.

**Comment by Raizo62** [comment]
I wanted capture strace of ubridge but honestly, i don't know if my test is correct because i have do anormal thing : i have started gns3 as root and replaced ubridge by bash script who do strace.

here log : 
[ubridge.log.txt](https://github.com/GNS3/gns3-gui/files/758345/ubridge.log.txt)

You can see : 

> 8864 send(5, 0x7f19a5ea9720, 8754, 0)            = -1
> 8864 perror("send")                              = <void>
> 8864 printf("Destination NIO listener thread for %s has stopped\n", "51c3e9fe-f892-47d9-b088-b6c7b3b6fa51-1") = 87
> 8864 pthread_exit(0, 0, 0, 74 <no return ...>
> 8864 +++ exited (status 0) +++
>

**Comment by noplay** [comment]
LiveRaizo is running in VirtualBox right? And eth0 is one of the virtualbox interface.

**Comment by Raizo62** [comment]
No. All my test with this bug are do on true PC. 
(I don't know why but Cloud device doesn't work when LiveRaizo is running in VirtualBox. Even with GNS3 1.5 of LiveRaizo v7. Then i must use LiveRaizo as router for my virtual network)

**Comment by Raizo62** [comment]
I am not agree with your new title "with virtualbox".

Any device connected to ethernet switch who is connected to Cloud Device lost connection (but yes, often, a virtualbox device is connected to this switch)

**Comment by noplay** [comment]
Do you have  VLAN on the network (virtual or physical)?

**Comment by noplay** [comment]
Are you using Jumbo frame or the network?

**Comment by noplay** [comment]
Could try recompiling ubridge and replace in nio.h:
```c
#define NIO_MAX_PKT_SIZE    2048
```

by 
```c
#define NIO_MAX_PKT_SIZE    16384
```

**Comment by Raizo62** [comment]
Yes, we have VLAN on physical network, but normally, my real PC don't see the trunk

"Jumbo frame on the network" : normally no

> Could try recompiling ubridge and replace in nio.h:

Yes, i can but this evening to test tomorrow :-(

**Comment by Raizo62** [comment]
I could tested only one time before the beginning of teaching in the classroom.

I have see the "send: Bad address" in the ubridge.log.

(I will retest because i had restart gns3 several times before good permissions on file. But i believe that will not change anything)

**Comment by Raizo62** [comment]
Second test : I confirm : I have see the "send: Bad address" in the ubridge.log.

**Comment by noplay** [comment]
@grossmj when you will debug ubridge for VMware perhaps you will see something for this one

**Comment by noplay** [comment]
Can you try the last version of ubridge? It's supposed to fix the issue.
https://github.com/GNS3/ubridge/releases/tag/v0.9.11

**Comment by Raizo62** [comment]
In the file "untitled/project-files/builtin/8f4bc703-bf5b-4b46-ac09-e43a8a3e6178/ubridge.log", i see a lot of : 
```
received frame is 26130 bytes (maximum is 10000 bytes)
received frame is 17442 bytes (maximum is 10000 bytes)
received frame is 18570 bytes (maximum is 10000 bytes)
received frame is 26130 bytes (maximum is 10000 bytes)
received frame is 23234 bytes (maximum is 10000 bytes)
received frame is 26130 bytes (maximum is 10000 bytes)
received frame is 22210 bytes (maximum is 10000 bytes)
received frame is 13554 bytes (maximum is 10000 bytes)
received frame is 13098 bytes (maximum is 10000 bytes)
received frame is 24682 bytes (maximum is 10000 bytes)
received frame is 27578 bytes (maximum is 10000 bytes)
received frame is 24682 bytes (maximum is 10000 bytes)
received frame is 17442 bytes (maximum is 10000 bytes)
received frame is 23234 bytes (maximum is 10000 bytes)
received frame is 24682 bytes (maximum is 10000 bytes)
received frame is 24682 bytes (maximum is 10000 bytes)
received frame is 24682 bytes (maximum is 10000 bytes)
received frame is 17058 bytes (maximum is 10000 bytes)
received frame is 26130 bytes (maximum is 10000 bytes)
received frame is 26130 bytes (maximum is 10000 bytes)
received frame is 21786 bytes (maximum is 10000 bytes)
received frame is 26130 bytes (maximum is 10000 bytes)
received frame is 20338 bytes (maximum is 10000 bytes)
received frame is 21786 bytes (maximum is 10000 bytes)
received frame is 26130 bytes (maximum is 10000 bytes)
received frame is 30474 bytes (maximum is 10000 bytes)
received frame is 24682 bytes (maximum is 10000 bytes)
received frame is 25826 bytes (maximum is 10000 bytes)
received frame is 29026 bytes (maximum is 10000 bytes)
```
But, the network seems to work again.
And i don't see "send: Bad address" but, must i see it again ?

**Comment by Raizo62** [comment]
The big size of received frames comes perhaps of network modification does to access to vdi by NFS :
The script who configures Live Raizo to use NFS does this :  
```
echo 262144 | sudo tee /proc/sys/net/core/rmem_default > /dev/null
echo 262144 | sudo tee /proc/sys/net/core/rmem_max > /dev/null
echo 262144 | sudo tee /proc/sys/net/core/wmem_default > /dev/null
echo 262144 | sudo tee /proc/sys/net/core/wmem_max > /dev/null
```

**Comment by Raizo62** [comment]
In the file, i have 4657 lines "received frame is"
The maximum value is 40290
I have only 153 values differents

**Comment by grossmj** [comment]
Well, that explains a lot! thanks for these details!

I don't know if we should handle these gigantic frames or not.

**Comment by grossmj** [comment]
We could bump the limit to 65536, that should take care of it.

**Comment by noplay** [comment]
And it should not cost a lot of ressources

**Comment by Raizo62** [comment]
I have tested without NFS : It seems that i don't have big frame.

But, what is worried me, now, is the size of this file ubridge.log. It can become very big. Can you remove the printf "received frame is" ? Or add option "--debug" used by GNS3 when we use this option ?

**Comment by noplay** [comment]
The received frame error is very important. This mean GNS3 can't handle the full network packet and you will lost data.

**Comment by Raizo62** [comment]
And if you increase the size to 65536, the file must not increase also.

**Comment by grossmj** [comment]
I increased to 65535. This will be fixed when we release uBridge 0.9.12.
